### PR TITLE
Support for single process single cell standalone test cases

### LINKED ISSUE #5118: MPAS-Seaice lacks single process idealized test cases
MPAS-Seaice lacks idealized single cell test cases for individual thermodynamic processes. This functionality would be useful for unit testing of the sea ice component processes

### LINKED ISSUE #5119: Sea ice conservation standalone tests dont mimic ocean interaction in E3SM
In standalone mode the sea ice model doesn't include the ocean mixed layer in conservation checks, and the mixed layer model represents the old POP ocean model, not the current MPAS-Ocean model interaction.

### PR DESCRIPTION
**Pull request by akturner**
Added several options for controlling which thermodynamic processes are active
Added option for conservation AM to include ocean heat, mass, energy content in analysis with mixed layer model
Added another mixed layer ocean model that mimics ocean model in E3SM for standalone testing
Added another forcing option to support idealized single cell test case

Fixes #5118
Fixes #5119

[NML]
[BFB]

### CODE REVIEW

File: components/mpas-seaice/src/shared/mpas_seaice_forcing.F
Comment (eclare108213) [inline components/mpas-seaice/src/shared/mpas_seaice_forcing.F:1190]: I think the nomenclature here is incorrect.  POP is a free-surface ocean model, although it is constant volume.  So the original ncar forcing data is usable for a "free surface" ocean model.

File: components/mpas-seaice/src/shared/mpas_seaice_column.F
Comment (eclare108213) [inline components/mpas-seaice/src/shared/mpas_seaice_column.F:5660]: Is there a (design) document somewhere that explains these differences in true detail?  We will need that for the Icepack documentation.

File: components/mpas-seaice/src/shared/mpas_seaice_column.F
Comment (eclare108213) [inline components/mpas-seaice/src/shared/mpas_seaice_column.F:1887]: What is the computational burden of adding 'if' conditionals for all of these individual processes?  They have to be parsed for each category, on each grid cell with ice, perhaps multiple times per timestep.

**Comment by rljacob** [comment]
@akturner please respond to review comments.

File: components/mpas-seaice/src/shared/mpas_seaice_column.F
Comment (akturner) [reply components/mpas-seaice/src/shared/mpas_seaice_column.F]: @eclare108213: Good point. I have recoded this so that there are no conditionals now, just multiplication by a real flag taking values (0,1). I will test performance of these changes as well

File: components/mpas-seaice/src/shared/mpas_seaice_forcing.F
Comment (akturner) [reply components/mpas-seaice/src/shared/mpas_seaice_forcing.F]: Good point, I have renamed the option to som_varying_thickness

File: components/mpas-seaice/src/shared/mpas_seaice_column.F
Comment (akturner) [reply components/mpas-seaice/src/shared/mpas_seaice_column.F:5650]: No design document. This is an additional subroutine in MPAS rather than the column package so doesn't necessarily need to be in icepack. Making this proxy was a good way to understand how frazil is currently done and I plan to try a better formulation so this may change soon anyway.

**Comment by akturner** [comment]
Single cell test case for idealized processes added here: https://github.com/MPAS-Dev/MPAS-Seaice_standalone_framework/pull/13

**Review by eclare108213 — APPROVED** [review]
Thanks for getting rid of the conditionals - this is a more elegant approach in my opinion and perhaps not as computationally impactful.  The title of this pull request does not fully represent what is in it -- the new conservation checks are particularly important.  I'm curious to know whether these tests have turned up anything surprising or needing to be fixed, but answering that question is not necessary for this PR.  Approving.

**Comment by rljacob** [comment]
telecon: waiting on review from @alicebarthel who is out of town.

**Comment by alicebarthel** [comment]
I ran a G-case 1-year test with this code. It ran successfully and the sea ice fluxes look consistent visually (see attached plots). My automated check technically failed but I think it may be an issue with the test itself. Looking into this now. 
seaice-ocean_fluxcomparison_cpl_am 20221016 test GMPAS-JRA singlecellsiPR TL319_EC30to60E2r2 anvil_hbudget_NH_1-1
seaice-ocean_fluxcomparison_cpl_am 20221016 test GMPAS-JRA singlecellsiPR TL319_EC30to60E2r2 anvil_hbudget_SH_1-1
seaice-ocean_fluxcomparison_cpl_am 20221016 test GMPAS-JRA singlecellsiPR TL319_EC30to60E2r2 anvil_hbudget_GL_1-1

**Comment by alicebarthel** [comment]
My only suggestion at this stage would be to change the streams.seaice file to replace: 
`clobber_mode="append"` with `clobber_mode="truncate"`
to match other streams (and avoid errors popping up when restarting/overwriting output files)

**Comment by eclare108213** [comment]
What are the units on the y-axes?  E.g. I'd expect the radiative fluxes to be ~50x larger in magnitude.

**Comment by alicebarthel** [comment]
@eclare108213 These fluxes are all in W/m2. Note that these are only the sea-ice ones. The atm/ocn radiative fluxes are much bigger (~+/-150 for net SW and ~+/-200-300 for LW). 
@akturner Can you remind us if there is area scaling: e.g. are we summing over the sea ice area then dividing by the whole hemisphere area? That could explain the order of magnitude difference. I have vague memories of discussing this with you a while back.

**Comment by akturner** [comment]
@alicebarthel: Which streams file are you referring to?

**Comment by eclare108213** [comment]
But they are for a single-cell column configuration.  Can they be made consistent with that view of the "world"?

**Comment by alicebarthel** [comment]
@akturner I meant the conservationCheck AM in streams.seaice. The code I checked out had them in "append" mode instead of "truncate" mode which means errors popped up when trying to overwrite the AM output files. Looking at the files, it does not seem to be something introduced by your changes, so I am happy to approve the PR.
@eclare108213 The test case I ran here is a "normal" (global) G-case, not the single-column mode. Adrian had asked me to check introducing these changes did not modify the standard e3sm flux check I have worked on.
@eclare108213 does that address your concern or do we need to double-check what a "hemispheric scaling" would look like in single-column mode? Is the idea to have the same sea-ice vs. coupler check for flux consistency in a single-column test? In that case, I don't know how the coupler would handle writing out the SH and NH fluxes...

**Comment by eclare108213** [comment]
I see, thanks for the clarification @alicebarthel.  Eventually we would like to look at MPAS-SI output in realistic magnitudes for at least one sea ice grid point in each hemisphere, but others are working on that problem now.

**Comment by rljacob** [comment]
telecon notes: needs a rebase and build system fixes.

**Comment by jonbob** [comment]
@akturner - the rebase and bld update look good. I'll test today and hopefully merge soon

**Comment by jonbob** [comment]
passes:
* SMS_D_Ld1.T62_oEC60to30v3.DTESTM.chrysalis_intel
* SMS_D_Ld1.ne30pg2_EC30to60E2r2.WCYCL1850.chrysalis_intel.allactive-wcprod
* ERS.ne11_oQU240.WCYCL1850NS.chrysalis_intel

with expected NML DIFFs

merged to next

**Comment by jonbob** [comment]
merged to master and expected NML DIFFs blessed

**Pull request merged**
